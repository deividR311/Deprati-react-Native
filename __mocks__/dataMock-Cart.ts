export const DataMock_Cart = {
  carts: [
    {
      appliedOrderPromotions: [],
      appliedProductPromotions: [],
      appliedVouchers: [],
      calculated: true,
      code: '4809002',
      deliveryAddress: {
        cellphoneNumber: '34949494',
        cellphonePreffix: '09',
        city: 'COT-287',
        country: {
          isocode: 'EC',
          name: 'Ecuador'
        },
        defaultAddress: true,
        email: 'alexis.enriquezca@gmail.com',
        firstName: 'Alejandra ',
        formattedAddress: 'Calle 6, Casa 23, Cotopaxi, Alaquez',
        id: '8819555172375',
        idNumber: '1721480299',
        idType: 'CEDULA',
        line1: 'Calle 6',
        line2: 'Casa 23',
        otherInfo: 'Laura',
        phone: '',
        region: {
          countryIso: 'EC',
          isocode: 'EC-COT',
          isocodeShort: '193',
          name: 'Cotopaxi'
        },
        shippingAddress: true,
        town: 'Alaquez',
        visibleInAddressBook: true
      },
      deliveryItemsQuantity: 1,
      deliveryOrderGroups: [
        {
          entries: [
            {
              basePrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              basePriceWithTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,60',
                priceType: 'BUY',
                value: 2.5984
              },
              basePriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              deliveryMode: {
                code: 'standard'
              },
              deliveryTimeRange: '5 a 7 días laborables',
              entryNumber: 0,
              giftPacking: false,
              pickup: false,
              product: {
                acceptGiftPacking: true,
                availableForPickup: false,
                baseOptions: [
                  {
                    selected: {
                      code: '226010190802030008',
                      priceData: {
                        currencyIso: 'USD',
                        formattedValue: '$2,32',
                        priceType: 'BUY',
                        value: 2.32
                      },
                      stock: {
                        stockLevel: 11,
                        stockLevelStatus: 'inStock'
                      },
                      url: '/camiseta-estampada/p/226010190802030008',
                      variantOptionQualifiers: []
                    },
                    variantType: 'GenericVariantProduct'
                  }
                ],
                baseProduct: '000226010190802030',
                categories: [
                  {
                    code: 'T8023',
                    name: '10',
                    parentCategoryName: 'Talla',
                    url: '/10/c/T8023'
                  },
                  {
                    code: 'C0110601',
                    name: 'Blanco',
                    parentCategoryName: 'Color',
                    url: '/blanco/c/C0110601'
                  },
                  {
                    code: '0701',
                    name: 'MUJERES',
                    url: '/mujeres/c/0701'
                  }
                ],
                code: '226010190802030008',
                deliveryModes: [
                  {
                    code: 'standard'
                  },
                  {
                    code: 'pickup'
                  }
                ],
                ean: '15282579',
                images: [
                  {
                    format: 'zoom',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
                  },
                  {
                    format: 'product',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
                  },
                  {
                    format: 'thumbnail',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
                  },
                  {
                    format: 'cartIcon',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
                  }
                ],
                isGiftProduct: false,
                isPromotionSpecialPrice: false,
                ivaCondition: '12%',
                name: 'CAMISETA ESTAMPADA',
                purchasable: true,
                stock: {
                  stockLevel: 11,
                  stockLevelStatus: 'inStock'
                },
                url: '/camiseta-estampada/p/226010190802030008'
              },
              quantity: 1,
              subtotal: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              totalDiscount: {
                currencyIso: 'USD',
                formattedValue: '$20,88',
                priceType: 'BUY',
                value: 20.88
              },
              totalPrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              totalPriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              updateable: true,
              volume: 0,
              weight: 0
            }
          ],
          totalPriceWithTax: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5999999999999996
          }
        }
      ],
      entries: [
        {
          basePrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          basePriceWithTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5984
          },
          basePriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          deliveryMode: {
            code: 'standard'
          },
          deliveryTimeRange: '5 a 7 días laborables',
          entryNumber: 0,
          giftPacking: false,
          pickup: false,
          product: {
            acceptGiftPacking: true,
            availableForPickup: false,
            baseOptions: [
              {
                selected: {
                  code: '226010190802030008',
                  priceData: {
                    currencyIso: 'USD',
                    formattedValue: '$2,32',
                    priceType: 'BUY',
                    value: 2.32
                  },
                  stock: {
                    stockLevel: 11,
                    stockLevelStatus: 'inStock'
                  },
                  url: '/camiseta-estampada/p/226010190802030008',
                  variantOptionQualifiers: []
                },
                variantType: 'GenericVariantProduct'
              }
            ],
            baseProduct: '000226010190802030',
            categories: [
              {
                code: 'T8023',
                name: '10',
                parentCategoryName: 'Talla',
                url: '/10/c/T8023'
              },
              {
                code: 'C0110601',
                name: 'Blanco',
                parentCategoryName: 'Color',
                url: '/blanco/c/C0110601'
              },
              {
                code: '0701',
                name: 'MUJERES',
                url: '/mujeres/c/0701'
              }
            ],
            code: '226010190802030008',
            deliveryModes: [
              {
                code: 'standard'
              },
              {
                code: 'pickup'
              }
            ],
            ean: '15282579',
            images: [
              {
                format: 'zoom',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
              },
              {
                format: 'product',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
              },
              {
                format: 'thumbnail',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
              },
              {
                format: 'cartIcon',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
              }
            ],
            isGiftProduct: false,
            isPromotionSpecialPrice: false,
            ivaCondition: '12%',
            name: 'CAMISETA ESTAMPADA',
            purchasable: true,
            stock: {
              stockLevel: 11,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-estampada/p/226010190802030008'
          },
          quantity: 1,
          subtotal: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          totalDiscount: {
            currencyIso: 'USD',
            formattedValue: '$20,88',
            priceType: 'BUY',
            value: 20.88
          },
          totalPrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          totalPriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          updateable: true,
          volume: 0,
          weight: 0
        }
      ],
      guid: '15901dbd-9966-4239-9a4d-fecc4f747589',
      net: true,
      orderDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      pickupItemsQuantity: 0,
      pickupOrderGroups: [],
      productDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      sellWithTaxes: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      sellWithoutTaxWithoutInterest: {
        currencyIso: 'USD',
        formattedValue: '$2,31',
        priceType: 'BUY',
        value: 2.31
      },
      sellWithoutTaxes: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      site: 'deprati',
      store: 'deprati',
      subTotal: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$20,88',
        priceType: 'BUY',
        value: 20.88
      },
      totalItems: 1,
      totalPrice: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalPriceWithTax: {
        currencyIso: 'USD',
        formattedValue: '$2,60',
        priceType: 'BUY',
        value: 2.6
      },
      totalTax: {
        currencyIso: 'USD',
        formattedValue: '$0,28',
        priceType: 'BUY',
        value: 0.28
      },
      totalVolume: 0,
      totalWeight: 0.25,
      user: {
        name: 'Alexis pruebas ENRIQUEZ',
        uid: 'alexis.enriquezca@gmail.com'
      },
      potentialOrderPromotions: [
        {
          consumedEntries: [],
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_A',
            description: 'Version 2 Hybris - PRUEBA APP',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        },
        {
          consumedEntries: [],
          description:
            'Completa $150 y  obten envío gratis. Aplica restricciones.',
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_C',
            description: 'Version 2 Hybris',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        }
      ],
      potentialProductPromotions: [],
      totalUnitCount: 1
    }
  ]
};
