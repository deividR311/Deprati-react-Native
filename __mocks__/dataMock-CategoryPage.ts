export const DataMock_CategoryContent = {
  defaultPage: false,
  name: 'Mujeres Category Page',
  template: 'CategoryPageTemplate',
  typeCode: 'CategoryPage',
  uid: 'MujeresCategoryPage',
  timestamp: 1679496809595,
  components: {
    MujeresFeatureComponent_1: {
      uid: 'MujeresFeatureComponent_1',
      typeCode: 'CategoryFeatureComponent',
      modifiedTime: '2023-02-27T07:59:47.012-05:00',
      name: 'MujeresFeatureComponent_1',
      container: 'false',
      media: {
        code: 'MujeresCategoryFeaturedBanner1',
        mime: 'image/jpeg',
        url: 'https://imagestg.deprati.com.ec/sys-master/images/h1c/hfb/8983274225694',
        downloadUrl:
          'https://imagestg.deprati.com.ec/sys-master/images/h1c/hfb/8983274225694'
      },
      title: 'Gafas Ray Ban Dorado',
      url: '/c/010401',
      childrenComponentsList: null,
      childrenComponents: {}
    },
    MujeresFeatureComponent_2: {
      uid: 'MujeresFeatureComponent_2',
      typeCode: 'CategoryFeatureComponent',
      modifiedTime: '2023-02-27T07:58:33.633-05:00',
      name: 'MujeresFeatureComponent_2',
      container: 'false',
      media: {
        code: 'MujeresCategoryFeaturedBanner2',
        mime: 'image/jpeg',
        url: 'https://imagestg.deprati.com.ec/sys-master/images/h77/h9a/8971632574494/categoryFeature2.jpg',
        downloadUrl:
          'https://imagestg.deprati.com.ec/sys-master/images/h77/h9a/8971632574494/categoryFeature2.jpg'
      },
      title: 'Gafas Ray Ban',
      url: '/c/010406?q=%3Arelevance%3Abrand%3A13353',
      childrenComponentsList: null,
      childrenComponents: {}
    },
    MujeresFeatureComponent_3: {
      uid: 'MujeresFeatureComponent_3',
      typeCode: 'CategoryFeatureComponent',
      modifiedTime: '2023-02-27T08:19:04.375-05:00',
      name: 'MujeresFeatureComponent_3',
      container: 'false',
      media: {
        code: 'MujeresCategoryFeaturedBanner3',
        mime: 'image/jpeg',
        url: 'https://imagestg.deprati.com.ec/sys-master/images/h88/h9d/8971632640030/categoryFeature3.jpg',
        downloadUrl:
          'https://imagestg.deprati.com.ec/sys-master/images/h88/h9d/8971632640030/categoryFeature3.jpg'
      },
      title: 'Bikinis',
      url: '/c/0106',
      childrenComponentsList: null,
      childrenComponents: {}
    },
    MujeresFeatureComponent_4: {
      uid: 'MujeresFeatureComponent_4',
      typeCode: 'CategoryFeatureComponent',
      modifiedTime: '2023-02-27T08:19:04.378-05:00',
      name: 'MujeresFeatureComponent_4',
      container: 'false',
      media: {
        code: 'MujeresCategoryFeaturedBanner4',
        mime: 'image/jpeg',
        url: 'https://imagestg.deprati.com.ec/sys-master/images/hdf/h9d/8971632672798/categoryFeature4.jpg',
        downloadUrl:
          'https://imagestg.deprati.com.ec/sys-master/images/hdf/h9d/8971632672798/categoryFeature4.jpg'
      },
      title: 'Traje de Baño Entero',
      url: '/c/0106?q=%3Aname-desc',
      childrenComponentsList: null,
      childrenComponents: {}
    },
    MujeresRotatingCategoryCarousel: {
      uid: 'MujeresRotatingCategoryCarousel',
      typeCode: 'RotatingImagesComponent',
      modifiedTime: '2023-01-25T11:34:19.338-05:00',
      name: 'DePrati Homepage Mobile Carousel',
      container: 'false',
      resolutionEnum: 'MOBILE',
      effect: 'FADE',
      banners:
        'MujeresBannerContainerPrimario_1_1 MujeresBannerContainerPrimario_2_1 MujeresBannerContainerPrimario_3_1',
      timeout: '0',
      childrenComponentsList: [
        'MujeresBannerContainerPrimario_1_1',
        'MujeresBannerContainerPrimario_2_1',
        'MujeresBannerContainerPrimario_3_1'
      ],
      childrenComponents: {
        MujeresBannerContainerPrimario_1_1: {
          uid: 'MujeresBannerContainerPrimario_1_1',
          typeCode: 'BannerComponent',
          modifiedTime: '2023-01-20T19:47:34+0000',
          name: 'Mujeres Banner Container Primario 1',
          container: 'false',
          external: 'false',
          media: {
            code: 'Mujeres-BannerPrimario_1_1_Mobile.jpg',
            mime: 'image/jpeg',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/h90/h8f/8971632214046/bannerPrimario1.jpg',
            downloadUrl:
              'https://imagestg.deprati.com.ec/sys-master/images/h90/h8f/8971632214046/bannerPrimario1.jpg'
          },
          headline: 'Tendencia Oasis',
          urlLink: '/corner/hyomujeres',
          childrenComponentsList: null,
          childrenComponents: {}
        },
        MujeresBannerContainerPrimario_2_1: {
          uid: 'MujeresBannerContainerPrimario_2_1',
          typeCode: 'BannerComponent',
          modifiedTime: '2022-10-28T16:01:52+0000',
          name: 'Mujeres Banner Container Primario 2',
          container: 'false',
          external: 'false',
          media: {
            code: 'Mujeres-BannerPrimario_2_1_Mobile.jpg',
            mime: 'image/jpeg',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/he7/h8f/8971632246814/bannerPrimario2.jpg',
            downloadUrl:
              'https://imagestg.deprati.com.ec/sys-master/images/he7/h8f/8971632246814/bannerPrimario2.jpg'
          },
          headline: 'Tendencia Simplcity',
          urlLink: '/c/0101',
          childrenComponentsList: null,
          childrenComponents: {}
        },
        MujeresBannerContainerPrimario_3_1: {
          uid: 'MujeresBannerContainerPrimario_3_1',
          typeCode: 'BannerComponent',
          modifiedTime: '2022-05-20T14:30:59+0000',
          name: 'Mujeres Banner Container Primario 3',
          container: 'false',
          external: 'false',
          media: {
            code: 'Mujeres-BannerPrimario_3_1_Mobile.jpg',
            mime: 'image/jpeg',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/he9/h8c/8971632181278/bannerPrimario3.jpg',
            downloadUrl:
              'https://imagestg.deprati.com.ec/sys-master/images/he9/h8c/8971632181278/bannerPrimario3.jpg'
          },
          headline: 'Vizzano',
          urlLink: '/c/0105?q=%3Aname-asc%3Abrand%3A11982',
          childrenComponentsList: null,
          childrenComponents: {}
        }
      }
    },
    MujeresCategoryFeatureCarousel: {
      uid: 'MujeresCategoryFeatureCarousel',
      typeCode: 'DepratiCategoryFeatureCarouselComponent',
      modifiedTime: '2023-03-14T15:28:21.264-05:00',
      name: 'Categorías principales',
      container: 'false',
      categoryList:
        'MujeresCarouselFeatureComponent_2 MujeresCarouselFeatureComponent_3 MujeresCarouselFeatureComponent_4 MujeresCarouselFeatureComponent_6',
      title: 'Categorías principales',
      childrenComponentsList: [
        'MujeresCarouselFeatureComponent_2',
        'MujeresCarouselFeatureComponent_3',
        'MujeresCarouselFeatureComponent_4',
        'MujeresCarouselFeatureComponent_6'
      ],
      childrenComponents: {
        MujeresCarouselFeatureComponent_2: {
          uid: 'MujeresCarouselFeatureComponent_2',
          typeCode: 'CategoryFeatureComponent',
          modifiedTime: '2022-05-20T14:30:59+0000',
          name: 'MujeresCarouselFeatureComponent_2',
          container: 'false',
          media: {
            code: 'MujeresCategoryCarouselFeaturedBanner2',
            mime: 'image/png',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/h09/hb0/8985954484254',
            downloadUrl:
              'https://imagestg.deprati.com.ec/sys-master/images/h09/hb0/8985954484254'
          },
          category: '0110',
          title: 'ROPA DEPORTIVA',
          childrenComponentsList: null,
          childrenComponents: {}
        },
        MujeresCarouselFeatureComponent_3: {
          uid: 'MujeresCarouselFeatureComponent_3',
          typeCode: 'CategoryFeatureComponent',
          modifiedTime: '2022-04-22T13:24:36+0000',
          name: 'MujeresCarouselFeatureComponent_3',
          container: 'false',
          media: {
            code: 'MujeresCategoryCarouselFeaturedBanner3',
            mime: 'image/png',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/h62/had/8985954517022',
            downloadUrl:
              'https://imagestg.deprati.com.ec/sys-master/images/h62/had/8985954517022'
          },
          category: '010201',
          title: 'BLUSAS Y CAMISEROS',
          childrenComponentsList: null,
          childrenComponents: {}
        },
        MujeresCarouselFeatureComponent_4: {
          uid: 'MujeresCarouselFeatureComponent_4',
          typeCode: 'CategoryFeatureComponent',
          modifiedTime: '2022-04-22T13:24:36+0000',
          name: 'MujeresCarouselFeatureComponent_4',
          container: 'false',
          media: {
            code: 'MujeresCategoryCarouselFeaturedBanner4',
            mime: 'image/png',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/h0b/had/8985954549790',
            downloadUrl:
              'https://imagestg.deprati.com.ec/sys-master/images/h0b/had/8985954549790'
          },
          category: '010104',
          title: 'VESTIDOS Y ENTERIZOS',
          childrenComponentsList: null,
          childrenComponents: {}
        },
        MujeresCarouselFeatureComponent_6: {
          uid: 'MujeresCarouselFeatureComponent_6',
          typeCode: 'CategoryFeatureComponent',
          modifiedTime: '2023-02-27T12:42:39+0000',
          name: 'MujeresCarouselFeatureComponent_6',
          container: 'false',
          media: {
            code: 'MujeresCategoryCarouselFeaturedBanner6',
            mime: 'image/png',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/ha1/hac/8985954582558',
            downloadUrl:
              'https://imagestg.deprati.com.ec/sys-master/images/ha1/hac/8985954582558'
          },
          category: '010401',
          title: 'CARTERAS',
          childrenComponentsList: null,
          childrenComponents: {}
        }
      }
    },
    MujeresCategoryBannerColecciones: {
      uid: 'MujeresCategoryBannerColecciones',
      typeCode: 'DepratiDoubleResponsiveBannerComponent',
      modifiedTime: '2023-01-25T15:40:19.812-05:00',
      name: 'Novedades',
      firstBannersList:
        'MujeresBannerContainerSecundario_1_1 MujeresBannerContainerSecundario_3_1',
      container: 'false',
      title: 'Novedades',
      childrenComponentsList: [
        'MujeresBannerContainerSecundario_1_1',
        'MujeresBannerContainerSecundario_3_1'
      ],
      childrenComponents: {
        MujeresBannerContainerSecundario_1_1: {
          uid: 'MujeresBannerContainerSecundario_1_1',
          typeCode: 'SimpleResponsiveBannerComponent',
          modifiedTime: '2022-05-24T21:13:37+0000',
          name: 'Componente_de_titular_adecuado_simple',
          container: 'false',
          media: {
            imageType: 'GALLERY',
            format: 'mobile',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/hb5/h99/8971632508958/bannerSecundario1.jpg',
            altText: 'Paraíso Creativo',
            width: '480'
          },
          urlLink: 'https://sstghybris.deprati.com.ec/es/gafas/c/010406',
          childrenComponentsList: null,
          childrenComponents: {}
        },
        MujeresBannerContainerSecundario_3_1: {
          uid: 'MujeresBannerContainerSecundario_3_1',
          typeCode: 'SimpleResponsiveBannerComponent',
          modifiedTime: '2023-02-27T13:19:04+0000',
          name: 'MujeresBannerContainerSecundario_3_1',
          container: 'false',
          media: {
            imageType: 'GALLERY',
            format: 'mobile',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/h99/h46/8983277666334/MujerepruebaEstef.jpg',
            width: '480'
          },
          urlLink: '/c/0106',
          childrenComponentsList: null,
          childrenComponents: {}
        }
      }
    },
    SiteLogoComponent: {
      uid: 'SiteLogoComponent',
      typeCode: 'SimpleBannerComponent',
      modifiedTime: '2022-05-20T11:47:01.945-05:00',
      name: 'Site Logo Component',
      container: 'false',
      external: 'false',
      media: {
        code: '/images/logo_deprati_responsive.svg',
        mime: 'image/svg+xml',
        altText: 'Deprati Storefront',
        url: 'https://imagestg.deprati.com.ec/sys-master/images/h9f/hfd/8969934569502',
        downloadUrl:
          'https://imagestg.deprati.com.ec/sys-master/images/h9f/hfd/8969934569502'
      },
      urlLink: '/',
      childrenComponentsList: null,
      childrenComponents: {}
    }
  },
  slots: {
    Section2: [
      'MujeresFeatureComponent_1',
      'MujeresFeatureComponent_2',
      'MujeresFeatureComponent_3',
      'MujeresFeatureComponent_4'
    ],
    Section1: [
      'MujeresRotatingCategoryCarousel',
      'MujeresCategoryFeatureCarousel',
      'MujeresCategoryBannerColecciones'
    ],
    SiteLogo: ['SiteLogoComponent'],
    SearchBox: [],
    NavigationBar: [],
    TopHeaderSlot: []
  }
};

export const DataMock_CategoryData = {
  imageType: 'GALLERY',
  format: 'mobile',
  url: 'https://imagestg.deprati.com.ec/sys-master/images/hc0/hcd/8985042092062',
  width: '480',
  link: {
    linkName: {},
    category: '01',
    external: 'false',
    target: 'false'
  },
  text: 'Mujeres',
  textColor: '#ffffff',
  order: '1'
};
