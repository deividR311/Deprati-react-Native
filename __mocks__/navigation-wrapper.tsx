import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
