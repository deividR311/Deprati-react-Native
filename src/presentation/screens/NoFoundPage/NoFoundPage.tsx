import { useNavigation } from '@react-navigation/native';
import React, { useLayoutEffect } from 'react';
import {
  StyleSheet,
  SafeAreaView,
  useWindowDimensions,
  Image,
  View
} from 'react-native';
import { NAV } from '../../../application/common';
import { COLORS } from '../../../application/common/colors';
import { MainButton } from '../../common-components/buttons/Button';

interface Props {
  headerShow?: boolean;
}

const NoFoundPage = (props: Props) => {
  const { headerShow = true } = props;
  const navigation = useNavigation();
  const { width, height } = useWindowDimensions();

  useLayoutEffect(() => {
    if (headerShow) {
      navigation?.setOptions({
        headerShown: true,
        headerTitle: 'Resultados de búsqueda',
        headerTitleAlign: 'left'
      });
    }
  }, [navigation]);

  //const widthButton = width * 0.7
  const aspectRatio = width / height;

  return (
    <SafeAreaView style={{ flex: 1 }}>
      <View style={[styles.container]}>
        <Image
          style={[styles.background]}
          imageStyle={{ aspectRatio }}
          resizeMode={'stretch'}
          source={require('../../../../assets/images/noResults.jpg')}
        />
        <MainButton
          onPress={() => {
            navigation.navigate(NAV.HOME);
          }}
          style={[styles.container_button]}
          title={'regresar a la página principal'.toLocaleUpperCase()}
        />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '90%',
    alignItems: 'center',
    backgroundColor: COLORS.WHITE
  },
  background: {
    width: '100%',
    minHeight: 500,
    maxHeight: 600,
    height: '100%',
    aspectRatio: 600 / 850,
    alignItems: 'center',
    justifyContent: 'flex-end',
    backgroundColor: COLORS.WHITE,
    marginBottom: 8
  },
  container_button: {
    width: '90%',
    marginHorizontal: 16,
    marginBottom: 16
  }
});

export default NoFoundPage;
