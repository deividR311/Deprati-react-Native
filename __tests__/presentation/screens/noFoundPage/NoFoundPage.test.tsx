import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../__mocks__/navigation-wrapper';
import NoFoundPage from '../../../../src/presentation/screens/NoFoundPage/NoFoundPage';

let screenTest: any;
describe('NoFoundPage', () => {
  it('renders correctly in Snapshot', () => {
    screenTest = render(<NavigationWrapper screen={NoFoundPage} />);
    expect(screenTest.toJSON()).toMatchSnapshot();
  });
});
