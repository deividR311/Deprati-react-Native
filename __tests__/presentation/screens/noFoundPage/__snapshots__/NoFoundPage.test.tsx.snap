// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoFoundPage renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="Resultados de búsqueda"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "height": "90%",
                  "width": "100%",
                },
              ]
            }
          >
            <Image
              imageStyle={
                Object {
                  "aspectRatio": 0.5622188905547226,
                }
              }
              resizeMode="stretch"
              source={
                Object {
                  "testUri": "../../../assets/images/noResults.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "aspectRatio": 0.7058823529411765,
                    "backgroundColor": "#ffffff",
                    "height": "100%",
                    "justifyContent": "flex-end",
                    "marginBottom": 8,
                    "maxHeight": 600,
                    "minHeight": 500,
                    "width": "100%",
                  },
                ]
              }
            />
            <View
              accessibilityLabel=""
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#E31A2C",
                  "borderRadius": 5,
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "marginHorizontal": 16,
                  "opacity": 1,
                  "paddingVertical": 10,
                  "width": "90%",
                }
              }
              testID=""
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": undefined,
                      },
                      Object {},
                      Object {},
                    ]
                  }
                >
                  REGRESAR A LA PÁGINA PRINCIPAL
                </Text>
              </View>
            </View>
          </View>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
